packages/ckeditor5-link/src/autolink.js

```diff
@@ -233,16 +233,16 @@ export default class AutoLink extends Plugin {
 		const model = this.editor.model;
 		const deletePlugin = this.editor.plugins.get( 'Delete' );
 
-		if ( !this.isEnabled || !isLinkAllowedOnRange( range, model ) ) {
+		const defaultProtocol = this.editor.config.get( 'link.defaultProtocol' );
+		const parsedUrl = addLinkProtocolIfApplicable( link, defaultProtocol );
+
+		if ( !this.isEnabled || !isLinkAllowedOnRange( range, model ) || !linkHasProtocol( parsedUrl ) ) {
 			return;
 		}
 
 		// Enqueue change to make undo step.
 		model.enqueueChange( writer => {
-			const defaultProtocol = this.editor.config.get( 'link.defaultProtocol' );
-			const parsedUrl = addLinkProtocolIfApplicable( link, defaultProtocol );
 			writer.setAttribute( 'linkHref', parsedUrl, range );
-
```

Comment:
Please don't remove existing empty lines :) (unless this was automated by IDE, btw. @DawidKossowski had the same thing).

Personally, I am a big fan of empty lines and separating code blocks. I think that in our codebase most of use like adding extra lines for readability reasons.

Reply:
Right, I am also a fan of separating code blocks, but in this case I've removed most of the code from the block so it made sense to me to merge both code blocks together. On the other hand it makes sense to visually separate actual command changes from broadcasting another, undo command.